### PR TITLE
[Feat][NSA] Implement a NSA compression forward kernel.

### PR DESCRIPTION
**Pull request by michaelwithu**
This PR introduces a Native Sparse Attention (NSA) compression forward operator, significantly enhancing the library's capabilities for efficient attention mechanisms. Key contributions include:
- [x] Core Implementation: A forward kernel for NSA compression implemented in TileLang, with full support for variable-length sequences (varlen);
- [x] Issue Resolution: Closes Issue #159 — "[Feature][NSA] Implement a NSA compression forward kernel";
- [x] Performance Validation: Adds NSACmpFwdVarlenBenchmark to quantitatively evaluate operator throughput and efficiency;
- [x] Correctness Assurance: Comprehensive unit tests (test_nsa_cmp_fwd_varlen_op) covering edge cases and numerical accuracy validation.

### CODE REVIEW

File: tests/ops/test_deepseek_nsa_cmp_fwd.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        "dim_v": dim_v,
+        "group": group,
+        "scale": scale,
+        "bc": bc,
+        "bs": bs,
+        "bk": bk,
+        "bv": bv,
+        "dtype": dtype,
+        "accum_dtype": accum_dtype,
+        "tune": tune,
```
[the remark this answers is not in the document]
Comment (michaelwithu) [reply tests/ops/test_deepseek_nsa_cmp_fwd.py]: take your advice.

File: top/ops/deepseek_nsa.py [hunk 2]
```diff
@@ -158,3 +159,57 @@ def forward(self, q: torch.Tensor, k: torch.Tensor, v: torch.Tensor,
...
+            "group": group,
+            "scale": scale,
+            "bc": bc,
+            "bs": bs,
+            "bk": bk,
+            "bv": bv,
+            "dtype": dtype,
+            "accum_dtype": accum_dtype,
+            "tune": tune,
+        }
```
[the remark this answers is not in the document]
Comment (michaelwithu) [reply top/ops/deepseek_nsa.py:201]: take your advice.
